### PR TITLE
Added support for LTC, ETH

### PR DESCRIPTION
**Pull request by appsdesh**
Generalized BTC oriented code to support all the coins supported non GDAX

### CODE REVIEW

File: schedule.go [hunk 1]
```diff
@@ -114,20 +117,22 @@ func (s *gdaxSchedule) Sync() error {
 	}
 
 	s.logger.Infow(
-		"Placing an order for BTC",
-		"purchaseCurrency", "USD",
+		"Placing an order for ", s.coin,
+		"purchaCurrency", "USD",
```
Comment (blampe) [inline schedule.go]: `purchaseCurrency`

File: schedule.go [hunk 2]
```diff
@@ -244,14 +249,14 @@ func (s *gdaxSchedule) timeSinceLastPurchase() (time.Duration, error) {
 	return now.Sub(lastTransactionTime), nil
 }
 
-func (s *gdaxSchedule) makePurchase() error {
+func (s *gdaxSchedule) makePurchase( productId string) error {
```
Comment (blampe) [inline schedule.go]: Nit: extra space in front of `productId`.

**Review by blampe — COMMENTED** [review]
Thank you for this!

File: main.go [hunk 3]
```diff
@@ -14,6 +14,11 @@ import (
 )
 
 var (
+	coin = kingpin.Flag(
```
Comment (blampe) [inline main.go:17]: Let's make this an [Enum](https://godoc.org/github.com/alecthomas/kingpin#ArgClause.Enum) to make sure the input is validated. 

I'd also like to keep this backwards-compatible, so let's preserve the existing behavior by having this default to "BTC" if the flag is unspecified.

Comment (appsdesh) [reply main.go:17]: done

**Pull request merged**